.artifakt/entrypoint.sh

```diff
@@ -29,6 +29,18 @@ su www-data -s /bin/bash -c '
 
 wait-for $APP_DATABASE_HOST:$APP_DATABASE_PORT --timeout=180
 
+
+lockdir="/data/docker-entrypoint.mutex"
+while ! mkdir "$lockdir"; do
+    echo 'Can not get lock. Waiting...' >&2
+    sleep 10
+done
+
+trap 'rmdir "$lockdir"; exit' EXIT INT TERM HUP
```

Comment:
Instead of `mkdir` for locking and avoiding race conditions you should use `flock` which is a command dedicated to managing process mutex, which will do all the "put a lock, if success do a command, if fails wait" mechanism in a single handy, dedicated, command 😃 

You can have more details [here](https://linux.die.net/man/1/flock)

Reply:
I went for "mkdir" instead of different options based on this article https://wiki.bash-hackers.org/howto/mutex

Now if you think "flock" is required, I will try another way

Reply:
@an0rak-dev switched to `flock` 🔐 PTAL